planner_cspace/include/grid_astar.h

```diff
@@ -166,88 +174,111 @@ class GridAstar
+      // Fetch tasks to be paralellized
       if (open_.size() < 1)
       {
         // No fesible path
-        // printf("No fesible path\n");
         if (return_best)
         {
           findPath(s, better, path);
         }
         return false;
       }
-      PriorityVec center = open_.top();
-      Vec p = center.v_;
-      float c = center.p_raw_;
-      float c_estim = center.p_;
-      open_.pop();
-      if (p == e || c_estim - c <= cost_leave)
+      bool found(false);
+      std::vector<PriorityVec> centers;
+      for (size_t i = 0; i < search_task_num_; ++i)
       {
-        e = p;
-        break;
+        if (open_.size() == 0)
+          break;
+        PriorityVec center = open_.top();
+        open_.pop();
+        if (center.v_ == e || center.p_ - center.p_raw_ <= cost_leave)
+        {
+          e = center.v_;
+          found = true;
+          break;
+        }
+        centers.push_back(center);
       }
+      if (found)
+        break;
 
-      float &gp = g[p];
-      if (c > gp)
-        continue;
-
-      if (c_estim - c < cost_estim_min)
+      for (auto it = centers.begin(); it < centers.end(); ++it)
       {
-        cost_estim_min = c_estim - c;
-        better = p;
-      }
+        const Vec p = it->v_;
+        const float c = it->p_raw_;
+        const float c_estim = it->p_;
+        float &gp = g[p];
+        if (c > gp)
+          continue;
 
-      auto tnow = boost::chrono::high_resolution_clock::now();
-      if (boost::chrono::duration<float>(tnow - ts).count() >= progress_interval)
-      {
-        std::list<Vec> path_tmp;
-        ts = tnow;
-        findPath(s, better, path_tmp);
-        cb_progress(path_tmp);
-      }
+        if (c_estim - c < cost_estim_min)
+        {
+          cost_estim_min = c_estim - c;
+          better = p;
+        }
 
-      std::vector<Vec> search_list = cb_search(p, s, e);
-      int updates = 0;
-      for (auto &diff : search_list)
-      {
-        Vec next = p + diff;
-        for (int i = NONCYCLIC; i < DIM; i++)
+        auto tnow = boost::chrono::high_resolution_clock::now();
+        if (boost::chrono::duration<float>(tnow - ts).count() >= progress_interval)
         {
-          next.cycleUnsigned(next[i], g.size()[i]);
+          std::list<Vec> path_tmp;
+          ts = tnow;
+          findPath(s, better, path_tmp);
+          cb_progress(path_tmp);
         }
-        if ((unsigned int)next[0] >= (unsigned int)g.size()[0] ||
-            (unsigned int)next[1] >= (unsigned int)g.size()[1])
-          continue;
-        if (g[next] < 0)
-          continue;
 
-        float cost_estim = cb_cost_estim(next, e);
-        if (cost_estim < 0 || cost_estim == FLT_MAX)
-          continue;
+        const std::vector<Vec> search_list = cb_search(p, s, e);
+        int updates = 0;
 
-        float cost = cb_cost(p, next, s, e);
-        if (cost < 0 || cost == FLT_MAX)
-          continue;
+#pragma omp parallel for schedule(static)
```

Comment:
not here but https://github.com/at-wat/neonavigation/pull/118/files#diff-edd86ae29de39b051a426a10b6dc22bfR209

Reply:
https://github.com/at-wat/neonavigation/pull/125